xcube/core/extract.py

```diff
@@ -314,38 +408,56 @@ def get_dataset_indexes(dataset: xr.Dataset,
     return interp_indexes
 
 
-def _empty_points_var(var: xr.DataArray, num_points):
+def _empty_points_var(var: xr.DataArray, num_points: int):
     fill_value = 0 if np.issubdtype(var.dtype, np.integer) else np.nan
     return xr.DataArray(np.full(num_points, fill_value, dtype=var.dtype),
                         dims=[INDEX_DIM_NAME], attrs=var.attrs)
 
 
-def _normalize_points(points: Union[xr.Dataset, pd.DataFrame, Mapping[str, Any]]) -> xr.Dataset:
-    if isinstance(points, pd.DataFrame):
-        points = xr.Dataset.from_dataframe(points)
-    elif not isinstance(points, xr.DataArray):
-        points = xr.Dataset(points)
-    return points
-
+def _normalize_series(
+        points: PointsLike,
+        col_names: Sequence[str],
+        force_dataset: bool = False,
+        param_name: str = "points"
+) -> Tuple[PointsLike, int]:
+    if not isinstance(points, (xr.Dataset, pd.DataFrame)):
```

Comment:
So in case the points are given as Mapping[str, SeriesLike] you will ignore them and create your own new points instead? This doesn't seem right to me. Please explain or, even better, write a test.

Reply:
Ah, wait. You don't ignore them. Still, please see my other comment and consider adding a test.

Reply:
Please read #434. This code block is already covered by tests. I'll add a comment.